setup/COSIGN.md

```diff
@@ -0,0 +1,98 @@
+# Using Connaisseur with Cosign signatures
+
+[Sigstore](https://sigstore.dev/) is a [Linux Foundation](https://linuxfoundation.org/) project that aims to provide public software signing and transparency to improve open source supply chain security. As part of the Sigstore project, [Cosign](https://github.com/sigstore/cosign) allows seamless container signing, verification and storage. You can read more about it [here](https://blog.sigstore.dev/cosign-signed-container-images-c1016862618?gi=95796f563c9f).
+
+As a first step, we have implemented the elementary function of verifying `cosign` generated signatures against the locally created corresponding public keys. We plan to expose further features of cosign in upcoming releases, so stay tuned!
+
+## Signing Container Images with Cosign
+
+Getting started with `cosign` is very well described in the [docs](https://github.com/sigstore/cosign). Please check there for detailed instructions. In short: After installation a keypair is generated via:
+
+```bash
+cosign generate-key-pair
+```
+
+You will be prompted to enter a password and a private (`cosign.key`) and public (`cosign.pub`) key are created. You can then use cosign to sign a container image using:
+
+```bash
+# Here, <IMAGE> is REPOSITORY/IMAGE_NAME:TAG
+cosign sign -key cosign.key <IMAGE>
+```
+
+You can then verify the created signature via:
+
+```bash
+cosign verify -key cosign.pub <IMAGE>
+```
+
+
+
+## Verification with Connaisseur
+
+Setting up Connaisseur (or short: `conny`) for `cosign` signatures only requires minor changes. In case of questions, please refer to the [default guide](README.md). In essence, you can just clone this repository:
+
+```bash
+git clone https://github.com/sse-secure-systems/connaisseur.git
+cd connaisseur
+```
+
+Next, configure `conny` to use cosign and the previously created public key for validation via the `helm/values.yaml`.  To do so, copy your `cosign.pub` key into `notary.rootPubkey`:
+
+```bash
+# the public part of the root key, for verifying notary's signatures
+  rootPubKey: |
+    -----BEGIN PUBLIC KEY-----
+    MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEUU2/8oNKPRTVjnTPY4WwUNFXYke3
+    CmpJpSaqBOFh92bdXZENv0AdR5qipqWxjGWO28Miyc2O6vGuoIyzzhQvuA==
+    -----END PUBLIC KEY-----
+```
+
+And set `notary.isCosign` to `true`:
+
+```bash
+  isCosign: true
+```
+
+You can then install `conny` via:
+
+```bash
+make install
+```
+
+
+
+## Test Signature Verification
+
+You can now test signature verification by deploying the signed image from above:
+
+```bash
+kubectl run signed --image=<IMAGE>
+```
+
+Or attempt to deploy another unsigned image.
+
+For the lazy ones, you can just configure our public key provided in the previous section and test with:
+
+```bash
+kubectl run signed --image=docker.io/xoph/ctest:signed
```

Comment:
We should move the image to the other hub before we release

Reply:
yes, mentioned it as a task in the PR

Reply:
ok :+1: I did see that but wanted to doublecheck whith regards to the time frame ;)

Reply:
it is done :rocket: